pynamodb/connection/base.py

```diff
@@ -361,12 +361,12 @@ def _make_api_call(self, operation_name, operation_kwargs):
                     timeout=self._request_timeout_seconds,
                     proxies=proxies,
                 )
+                response.raise_for_status()
```

Comment:
So we actually seem to turn failure status codes to pynamodb exceptions by parsing DynamoDB error responses. The "non-JSON content injected by middleware" case is the outlier here.

You can try rewriting as:
```py
try:
  ...
  try:
    data = request.json()
  except ValueError:
    if not response:
      response.raise_for_status()
    else:
      raise  # for the unlikely case that we got 2xx with unparsable JSON
except requests.RequestException as e:
  ...
```